### PR TITLE
Add concurrent indexing/querying challenge to PMC track

### PR DESCRIPTION
**Pull request by DJRickyB**

### CODE REVIEW

File: pmc/challenges/indexing-querying.json [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+      "clients": {{bulk_indexing_clients | default(8)}},
+      "ignore-response-error-level": "{{error_level | default('non-fatal')}}"
+    },
+    {
+      "name": "refresh-after-index",
+      "operation": "refresh"
+    },
+    {
+      "operation": {
+        "operation-type": "force-merge",
```
Comment (ebadyano) [inline pmc/challenges/indexing-querying.json:54]: did you mean to also `force_merge_max_num_segments` to this new challenge?

File: pmc/challenges/indexing-querying.json [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    },
+    {
+      "parallel": {
+        "tasks": [
+          {
+            "name": "concurrent-index-append",
+            "operation": {
+              "operation-type": "bulk",
+              "bulk-size": {{bulk_size | default(500)}},
+              "ingest-percentage": 50
```
Comment (ebadyano) [inline pmc/challenges/indexing-querying.json:83]: do we only want to index half of the data?

Comment (b-deam) [reply pmc/challenges/indexing-querying.json:83]: I think this makes sense, we ingest the first half here:
https://github.com/elastic/rally-tracks/blob/ee2a1f06940b62f34a696a6a0bebf32e7e25993d/nyc_taxis/challenges/indexing-querying.json#L32-L37

Then during querying, we ingest the remaining 50%:
https://github.com/elastic/rally-tracks/blob/ee2a1f06940b62f34a696a6a0bebf32e7e25993d/nyc_taxis/challenges/indexing-querying.json#L61-L82

**Comment by DJRickyB** [comment]
I originally thought that might be the case as well but i was mistaken.
Same 50% each time, could probably put it up to 100 but this is what I've
been using in recent benchmarks

File: pmc/challenges/default.json [hunk 3]
```diff
@@ -143,6 +143,9 @@
         {
           "operation": {
             "operation-type": "force-merge",
+{%- if force_merge_max_num_segments is defined %}
```
Comment (ebadyano) [inline pmc/challenges/default.json:146]: I think this added force merge max segments to `append-no-conflicts-index-only` challenge, but not the new one

**Review by ebadyano — APPROVED** [review]
lgtm

**Review by b-deam — COMMENTED** [review]
LGTM 2️⃣

**Pull request merged**